File: src/main/java/com/ecoroute/backend/domain/model/Order.java
package com.ecoroute.backend.domain.model;

import java.time.OffsetDateTime;

public record Order(
    Long id,
    String trackingNumber,
    String externalReference,
    Long routeId,
    OrderStatus status,
    
    // Recipient Info
    String recipientName,
    String recipientPhone,
    String recipientEmail,
    
    // Delivery Address
    String deliveryAddress,
    String deliveryCity,
    String deliveryDistrict,
    Double latitude,
    Double longitude,
    
    // Constraints
    Integer priority,
    OffsetDateTime estimatedDeliveryWindowStart,
    OffsetDateTime estimatedDeliveryWindowEnd,
    
    OffsetDateTime createdAt,
    OffsetDateTime updatedAt
) {}

File: mobile-app/lib/features/orders/domain/entities/order.dart
class OrderEntity {
  final int id;
  final String trackingNumber;
  final int routeId;
  final String status;
  final String recipientName;
  final String deliveryAddress;
  final double? latitude;
  final double? longitude;

  OrderEntity({
    required this.id,
    required this.trackingNumber,
    required this.routeId,
    required this.status,
    required this.recipientName,
    required this.deliveryAddress,
    this.latitude,
    this.longitude,
  });
}
